src/librustc_metadata/schema.rs

```diff
@@ -190,6 +199,20 @@ pub struct CrateRoot<'tcx> {
     pub profiler_runtime: bool,
     pub sanitizer_runtime: bool,
     pub symbol_mangling_version: SymbolManglingVersion,
+
+
+}
+
+/// Stores extra information about proc macros
+/// that would be difficult to store along with
+/// the dylib symbol.
+/// Currently, this is only used by rustdoc,
+/// which needs access to the proc macro span and
+/// attributes for inlining purposes
```

Comment:
A bit suboptimal, but I guess using this for expansion as well would be a bit more work?